ls1intum/Artemis
# Feature/programming exercise/test case build trigger

### Checklist

### Motivation and Context



Fixes issue with a client implementation: https://github.com/ls1intum/Artemis/issues/771
Issue is not resolved before a successive server implementation (see description).

### Description


Only client side implementation: When a test case was changed in the test case table, the user is notified that he should probably trigger the build of all submissions:
- Change guard opens a confirm if the test cases were saved
- Warning indicators are shown and the build trigger button is enabled (see screenshots)

In future we might also want to save submission runs in the database somehow so we can detect a "dirty" submission state when loading the table.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration > Programming Exercise > Manage Test Cases
3. Edit a test case, e.g. by changing a weight
4. Save the test cases
5. Try to leave the page - should not be possible without confirm
6. Trigger all and wait for the process to finish, then leave the page (be careful to not use an exercise with a lot of participations here ;))

### Screenshots



![image](https://user-images.githubusercontent.com/28230611/64525219-a1d40c00-d300-11e9-8d5f-bc95dca363b1.png)

**Change guard:**

![image](https://user-images.githubusercontent.com/28230611/64846349-bd932700-d60c-11e9-9df9-a10e72cfe1f7.png)
